src/controllers/update/beneficial.owner.type.controller.ts

```diff
@@ -22,6 +22,43 @@ import { checkEntityRequiresTrusts, getTrustLandingUrl } from "../../utils/trust
 import { retrieveTrustData } from "../../utils/update/trust.model.fetch";
 import { saveAndContinue } from "../../utils/save.and.continue";
 import { Session } from "@companieshouse/node-session-handler";
+import { validationResult } from "express-validator/src/validation-result";
+import { FormattedValidationErrors, formatValidationError } from "../../middleware/validation.middleware";
+
+type BeneficialOwnerTypePageProperties = {
+  backLinkUrl: string;
+  templateName: string;
+  errors?: FormattedValidationErrors;
+  hasExistingBosMos: boolean;
+  hasNewlyAddedBosMos: boolean;
+};
+
+const getPageProperties = (
+  req: Request,
+  errors?: FormattedValidationErrors,
+): BeneficialOwnerTypePageProperties => {
+  const appData: ApplicationData = getApplicationData(req.session);
+
+  const allBosMos = [
+    ...(appData[BeneficialOwnerIndividualKey] ?? []),
+    ...(appData[BeneficialOwnerOtherKey] ?? []),
+    ...(appData[BeneficialOwnerGovKey] ?? []),
+    ...(appData[ManagingOfficerCorporateKey] ?? []),
+    ...(appData[ManagingOfficerKey] ?? [])
+  ];
+
+  const hasNewlyAddedBosMos = allBosMos.find(boMo => !boMo.ch_reference) !== undefined;
+  const hasExistingBosMos = allBosMos.find(boMo => boMo.ch_reference) !== undefined;
+
+  return {
+    backLinkUrl: config.UPDATE_BENEFICIAL_OWNER_BO_MO_REVIEW_URL,
+    templateName: config.UPDATE_BENEFICIAL_OWNER_TYPE_PAGE,
+    hasNewlyAddedBosMos,
+    hasExistingBosMos,
+    ...appData,
+    errors
+  };
```

Comment:
I would do some manual checks to make sure the Remove journey still works ok. there is a section in the checkValidations that re-set the 'journey' page param - https://github.com/companieshouse/overseas-entities-web/blob/main/src/middleware/validation.middleware.ts#L62

it might be ok if we are passing appData in, but i would test it in docker to make sure it is ok, eg does it loose the 'Remove' page title if you get a validation error on that page on a Remove journey, and also clicking 'sign out' after a validation error on Remove journey on that page, then clicking the back link etc..

Reply:
Yes, I'm still seeing "Apply to remove an overseas entity from the register" as the page title after triggering an error with these changes